plugins/services/src/js/components/dsl/FuzzyTextDSLSection.js

```diff
@@ -33,7 +33,12 @@ class FuzzyTextDSLSection extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     const { expression } = nextProps;
-    const data = DSLUtil.getPartValues(expression, EXPRESSION_PARTS);
+    let data = DSLUtil.getPartValues(expression, EXPRESSION_PARTS);
+
+    if (!data || !data.text) {
+      data = { text: "" };
```

Comment:
If the text is not defined it doesn't reset the input field to an empty value when we reset the filters.

Reply:
Hm, ok. Then let's move that part to the util so that the util always provides correct and wellformed data.

Reply:
I updated the check in the input since it is needed only over there.

I also responded to the comment for the shouldComponentUpdate over here => https://github.com/dcos/dcos-ui/pull/2472#discussion_r158072691